test/aria/widgets/container/dialog/sizes/DialogMaxWidthTestCase.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2012 Amadeus s.a.s.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Test for issue #10
+ */
+Aria.classDefinition({
+    $classpath : "test.aria.widgets.container.dialog.sizes.DialogMaxWidthTestCase",
+    $extends : "aria.jsunit.TemplateTestCase",
+    $prototype : {
+
+        checkSizes : function(dialogId, mainWidth) {
+            var dialog = this.dialogs[dialogId];
+
+            var dom = dialog._domElt;
+            var children = dom.children;
+            var dlgBody = children[0];
+            var dlgTitle = children[1];
+            var dlgTitleText = aria.utils.Dom.getElementsByClassName(dlgTitle, "sDialog_title");
+            var dlgCloseIcon = aria.utils.Dom.getElementsByClassName(dlgTitle, "sDialog_close");
+
+            this.assertEqualsWithTolerance(dlgBody.offsetWidth, mainWidth, 8, dialogId + " width should be %2 instead of %1");
+            this.assertEqualsWithTolerance(dlgTitle.offsetWidth, mainWidth - 10, 8, dialogId + " title width should be %2 instead of %1");
+            this.assertEquals(dlgTitleText.offsetTop, dlgCloseIcon.offsetTop, "The close icon should be on teh first line");
+        },
+
+
+        runTemplateTest : function () {
+            // The main template is Ready even if it contains a template inside the Dialog, wait
+
+            var dialogIds = ["dialog1", "dialog2", "dialog3", "dialog4"];
+            this.waitFor({
+                condition : function() {
+
+                    // Wait for every dialog ready and open
+                    var dialogs = this.dialogs = {};
+                    for (var i = 0; i < dialogIds.length; i += 1) {
+                        var id = dialogIds[i];
+                        var currentInstance = dialogs[dialogIds[i]] = this.getWidgetInstance(dialogIds[i]);
```

Comment:
It would be better to use `id` instead of `dialogIds[i]`, otherwise, what's the point of creating the `id` variable?